libjcat/jcat-self-test.c

```diff
@@ -634,6 +637,33 @@ jcat_ed25519_engine_func(void)
 	g_assert_error(error, G_IO_ERROR, G_IO_ERROR_INVALID_DATA);
 	g_assert_null(result_fail);
 	g_clear_error(&error);
+
+	/* verify signing with an imported key */
+	blob_sig1 = jcat_engine_self_sign(engine, data_fwbin, JCAT_SIGN_FLAG_NONE, &error);
+	g_assert_no_error(error);
+	g_assert_nonnull(blob_sig1);
+	result_pass = jcat_engine_self_verify(engine,
```

Comment:
i think this needs to be `result_pass2` and `g_autoptr`'d? no?

Reply:
Sorry, I didn't realize that there is a separate test (`jcat_ed25519_engine_self_signed_func`) exercising this (but only with non-existent keys). I've extended the test instead so there shouldn't be a memleak.